### PR TITLE
refactor!: reorganize bones-bevy-renderer.

### PR DESCRIPTION
**Pull request by zicklag**
Submitting on behalf of @tekhaeraav:

This is a reorganization of bones-bevy-renderer that cleans up some dead code and attempts to make the renderer easier to understand without making any breaking changes.

### CODE REVIEW

**Review by zicklag — COMMENTED** [review]
This needs to be tested with Jumpy to make sure there's no regressions, but otherwise I approve ✅.

There's one added feature that lets you start the game without waiting for assets to load, in case you want to handle asset loading yourself ( a use-case in the Bones game maker UI @TekhnaeRaav  is working on ).

The rest is minimally changed functionally, mostly re-organized.

**Review by MaxCWhitehead — APPROVED** [review]
Looks great thank you @TekhnaeRaav!

**Comment by MaxCWhitehead** [comment]
Once get CI errors cleaned up + confirm works on jumpy can merge.

File: framework_crates/bones_bevy_renderer/src/storage.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                .open(&self.storage_path)
+                .expect("Failed to open storage file");
+            let mut serializer = serde_yaml::Serializer::new(file);
+            LoadedStorage(data)
+                .serialize(&mut serializer)
+                .expect("Failed to serialize to storage file.");
+        }
+
+        fn load(&mut self) -> Vec<SchemaBox> {
+            use anyhow::Context;
```
Comment (zicklag) [inline framework_crates/bones_bevy_renderer/src/storage.rs:85]:
```suggestion
```

**Comment by MaxCWhitehead** [comment]
Seems to work with jumpy - just had to define the preload setting.

**Pull request merged**